pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -107,9 +108,46 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
+func (r ReplicaSetReconciler) ensureAutomationConfig(mdb mdbv1.MongoDB) error {
+	cm, err := buildAutomationConfigConfigMap(mdb)
+	if err != nil {
+		return err
+	}
+	if err := r.cmClient.CreateOrUpdate(cm); err != nil {
+		return err
+	}
+	return nil
+}
+
+func buildAutomationConfig(mdb mdbv1.MongoDB) automationconfig.AutomationConfig {
+	domain := getDomain(mdb.ServiceName(), mdb.Namespace, mdb.Name)
+	return automationconfig.NewBuilder().
+		SetTopology(automationconfig.ReplicaSetTopology).
+		SetName(mdb.Name).
+		SetDomain(domain).
+		SetMembers(mdb.Spec.Members).
+		SetMongoDBVersion(mdb.Spec.Version).
+		SetAutomationConfigVersion(1). // TODO: Correctly set the version
+		Build()
+}
+
+func buildAutomationConfigConfigMap(mdb mdbv1.MongoDB) (corev1.ConfigMap, error) {
+	ac := buildAutomationConfig(mdb)
+	acBytes, err := json.Marshal(ac)
+	if err != nil {
+		return corev1.ConfigMap{}, err
+	}
+
+	return configmap.Builder().
+		SetName(mdb.ConfigMapName()).
+		SetNamespace(mdb.Namespace).
+		SetField(automationConfigKey, string(acBytes)).
+		Build(), nil
+}
+
 func getDomain(service, namespace, clusterName string) string {
```

Comment:
do we want to export this function so that we can use it in enterprise package?

![](https://media.giphy.com/media/8HsfVysBeRx7O/giphy.gif)

Reply:
this small function might be overkill, but it could worth while. I think we should wait until we have more relevant functions that can be grouped into a package before we export it though.